kernel/constr.ml

```diff
@@ -1289,6 +1290,14 @@ let hashcons (sh_sort,sh_ci,sh_construct,sh_ind,sh_con,sh_na,sh_id) =
     let h = !accu land 0x3FFFFFFF in
     (HashsetTermArray.repr h t term_array_table, h)
 
+  and hash_list_array l =
+    let fold accu c =
+      let c, h = sh_rec c in
+      (combine accu h, c)
+    in
+    let h, l = List.fold_left_map fold 0 l in
+    (l, h land 0x3FFFFFFF)
```

Comment:
Shouldn't you put the list in a hashtable?

Reply:
There is a comment above: there is no point in hashconsing evars, they never reach past kernel typing.

Reply:
Shouldn't it be error "kernel doesn't support evars" then?

Reply:
It doesn't hurt to be a bit more lenient, we might have users fancying hashconsing random terms.

Reply:
So let's put the list in a table then?

Reply:
Nah, just let it be that way...